File: TwoSumI.swift
func twoSum(nums: [Int], _ target: Int) -> [Int] {
    var dict = [Int: [Int]]()
    var result = [-1, -1]
    
    for var index = 0; index < nums.count; index++ {
        
        if let _ = dict[nums[index]] {
            dict[nums[index]]!.append(index)
        } else {
            dict[nums[index]] = [index]
        }
    }
    
    for (key, _) in dict {
        
        let delta = target - key
        //found one
        if let indices = dict[delta] {
            if delta == key && indices.count >= 2{
                result = [indices[0] + 1, indices[1] + 1]
            } else if delta != key {
                result = dict[key]![0] < dict[delta]![0] ? [dict[key]![0] + 1, dict[delta]![0] + 1] : [dict[delta]![0] + 1, dict[key]![0] + 1]
            }
        }
    }
    return result
}

File: Leetcode/TwoSumI.java
public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        result[0] = -1;
        result[1] = -1;
        if (nums == null || nums.length < 2) {
            return result;
        }
        HashMap<Integer, Integer> dict = new HashMap<Integer, Integer>();
        for (int i =0; i < nums.length; i++) {
            int delta = target - nums[i];
            if (dict.containsKey(delta)) {
                result[0] = dict.get(delta);
                result[1] = i;
                return result;
            }
            dict.put(nums[i], i);
        }
        return result;
    }
